### ISSUE TITLE
[TextBox] Using custom user agent causes a TypeError when entering value

### ISSUE THREAD

**Issue by kkopec** [telerik/kendo-angular#3622]
**Describe the bug**
Overriding user agent string to a non standard value causes a TypeError when a value is entered in a `TextBox` component.

`kendo-angular-inputs` uses a safari browser check in Textbox's `handleInput` method that is left unguarded against null chaining:
```
const isSafari = (userAgent) => {
    return detectDesktopBrowser(userAgent).safari ||
        (detectMobileOS(userAgent) && detectMobileOS(userAgent).browser === 'mobilesafari');
```
The first condition does not verify if the result of `detectDesktopBrowser` is valid and errors on null causing a:
```TypeError: Cannot read properties of null (reading 'safari')```

**To Reproduce**
1. Go to [this stackblitz example](https://stackblitz.com/edit/angular-ebc5de)
2. Open devtools
3. At the bottom click three dots (More Tools, next to Console) and open Network conditions
4. Change User agent to a non-existent custom value, e.g. `qwerty`
5. Type anything in a textbox

**Expected behavior**
Textbox should work properly regardless of custom user agent overrides.

**Additional context**
The bug was introduced in v7.5.2.

**Comment by Raisolution** [comment]
Hi @kkopec 

The mentioned operation is a custom modification, which is otherwise not allowed in a real-world application. The [userAgent field](https://developer.mozilla.org/en-US/docs/Web/API/Navigator/userAgent) is a **read-only** property and we rely on it on multiple places in our code base to determine the current browser.

More information on the browsers supported by Kendo UI for Angular is available here:
https://www.telerik.com/kendo-angular-ui/components/installation/requirements/#toc-browsers

**Comment by Raisolution** [comment]
Regarding the linked part of the TextBox component, we will update the mentioned if condition to address the problem in question.

**Comment by Raisolution** [comment]
Fix available in `v8.0.9` of the Inputs package.
